components/cliniko/cliniko.app.js

```diff
@@ -0,0 +1,63 @@
+const { axios } = require("@pipedreamhq/platform");
+
+module.exports = {
+  type: "app",
+  app: "cliniko",
+  propDefinitions: {
+    patientId: {
+      type: "integer",
+      label: "Patient ID",
+      description: "Enter a unique patient ID.",
+    },
+  },
+  methods: {
+    /**
+     * Get the Cliniko API key for the authenticated user.
+     * @returns {String} The Cliniko API key.
+     */
+    _apiKey() {
+      return this.$auth.api_key;
+    },
+    _shard() {
+      return this.$auth.shard;
+    },
+    _baseApiUrl() {
+      return `https://api.${this._shard()}.cliniko.com/v1`;
+    },
+    _makeRequestConfig(url) {
+      const auth = {
+        username: this._apiKey(),
+        password: "", // No password needed.
+      };
+      // Cliniko requires contact email address in User-Agent header.
+      // See https://github.com/redguava/cliniko-api#identifying-your-application.
+      const headers = {
+        "Accept": "application/json",
+        "User-Agent": "Pipedream (cliniko@pipedream.com)",
```

Comment:
I like this pattern but "cliniko@pipedream.com" won't resolve to an address we monitor. For now we set the user agent to:

```
@PipedreamHQ/pipedream v0.1
```

in all components